### PR TITLE
feat(nutanix): add node dashboard and tiles

### PR DESCRIPTION
**Pull request by varun257**
| Question         | Answer
| ---------------- | ---
| Branch?          | `feat/nutanix`
| Bug fix?         | no
| New feature?     | yes
| Breaking change? | no
| Tickets          | Fix #MANAGER-7450, #MANAGER-7451
| License          | BSD 3-Clause

## Description

Add server and hardware tile to view a dedicated server in nutanix application

### CODE REVIEW

File: packages/manager/apps/nutanix/package.json [hunk 1]
```diff
@@ -19,6 +19,7 @@
     "start:watch": "lerna exec --stream --parallel --scope='@ovh-ux/manager-nutanix-app' --include-dependencies -- npm run dev:watch --if-present"
   },
   "dependencies": {
+    "@ovh-ux/manager-bm-server-components": "^0.0.0 || ^1.0.0",
```
Comment (marie-j) [inline packages/manager/apps/nutanix/package.json:22]: Seems like some dependencies are missing here, `@ovh-ux/ng-at-internert`, `@ovh-ux/ng-ovh-feature-flipping` required by this module

File: packages/manager/modules/nutanix/src/dashboard/nodes/node/general-info/template.html [hunk 2]
```diff
@@ -0,0 +1,21 @@
+<div>
+    <div class="row d-md-flex mb-5">
```
Comment (marie-j) [inline packages/manager/modules/nutanix/src/dashboard/nodes/node/general-info/template.html:2]: Is the first `div` necessary ?

File: packages/manager/modules/nutanix/src/dashboard/nodes/node/general-info/translations/Messages_fr_FR.json [hunk 3]
```diff
@@ -0,0 +1,17 @@
+{
+  "server_datacenter_ERI": "London (ERI {{number}}) - England",
```
Comment (marie-j) [inline packages/manager/modules/nutanix/src/dashboard/nodes/node/general-info/translations/Messages_fr_FR.json:2]: Maybe the `Region` service could be reused for this purpose (https://github.com/ovh/manager/tree/develop/packages/manager/modules/manager-components/src/region)

File: packages/manager/modules/nutanix/src/dashboard/nodes/node/general-info/edit-display-name/controller.js [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+  }
+
+  onUpdateDisplayNameFormSubmit() {
+    if (!this.actionEnabled) {
+      return this.goBack();
+    }
+
+    this.loading.update = true;
+
+    return this.Nutanix.updateDisplayName({
```
Comment (radireddy) [inline packages/manager/modules/nutanix/src/dashboard/nodes/node/general-info/edit-display-name/controller.js:15]: You are not updating Nutanix cluster name here. You are updating Nutanix node name which is dedicated server. Please reuse DedicatedService. Lets have a discussion on this.

File: packages/manager/modules/nutanix/src/dashboard/nodes/node/general-info/edit-display-name/controller.js [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+      serviceId: this.nodeId,
+      serviceName: this.dedicatedServer.name,
+      displayName: this.displayName,
+    })
+      .then(() => {
+        this.goBack(
+          this.$translate.instant(
+            'nutanix_server_displaname_edit_displayname_success',
```
Comment (radireddy) [inline packages/manager/modules/nutanix/src/dashboard/nodes/node/general-info/edit-display-name/controller.js:23]:
```suggestion
            'nutanix_node_general_info_edit_display_name_success',
```

File: packages/manager/modules/nutanix/src/dashboard/nodes/node/general-info/edit-display-name/controller.js [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+          ),
+        );
+      })
+      .catch(({ data }) => {
+        this.goBack(
+          this.$translate.instant(
+            'nutanix_server_displaname_edit_displayname_error',
```
Comment (radireddy) [inline packages/manager/modules/nutanix/src/dashboard/nodes/node/general-info/edit-display-name/controller.js:30]:
```suggestion
            'nutanix_node_general_info_edit_display_name_error',
```
please use 'nutanix_node' prefix for all node related translations.

File: packages/manager/modules/nutanix/src/dashboard/nodes/node/general-info/edit-display-name/controller.js [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+
+  $onInit() {
+    this.loading = {
+      init: false,
+      update: false,
+    };
+    this.actionEnabled = true;
+
+    this.loading.init = true;
+    this.Nutanix.getServer(this.nodeId)
```
Comment (radireddy) [inline packages/manager/modules/nutanix/src/dashboard/nodes/node/general-info/edit-display-name/controller.js:49]: This need to move to DedicatedService. You need to reuse existing DedicatedService.

File: packages/manager/modules/nutanix/src/dashboard/nodes/node/general-info/edit-display-name/template.html [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+    data-ng-submit="$ctrl.onUpdateDisplayNameFormSubmit()"
+>
+    <oui-modal
+        data-primary-label="{{($ctrl.actionEnabled ? 'common_change' : 'global_OK') | translate }}"
+        data-secondary-label="{{$ctrl.actionEnabled ? ('common_cancel' | translate) : '' }}"
+        data-secondary-action="$ctrl.$state.go('^')"
+        data-on-dismiss="$ctrl.$state.go('^')"
+        data-loading="$ctrl.loading.init || $ctrl.loading.update"
+        data-type="{{ !$ctrl.actionEnabled ? 'warning' : '' }}"
+        data-heading="{{ :: 'nutanix_server_displaname_edit_title' | translate }}"
```
Comment (radireddy) [inline packages/manager/modules/nutanix/src/dashboard/nodes/node/general-info/edit-display-name/template.html:13]: add 'nutanix_node' prefix

File: packages/manager/modules/nutanix/src/dashboard/nodes/node/general-info/edit-display-name/translations/Messages_fr_FR.json [hunk 9]
```diff
@@ -0,0 +1,7 @@
+{
+  "nutanix_server_displaname_edit_title": "Modifier le nom",
```
Comment (radireddy) [inline packages/manager/modules/nutanix/src/dashboard/nodes/node/general-info/edit-display-name/translations/Messages_fr_FR.json:2]: change 'nutanix_server' to 'nutanix_node' for all translations

File: packages/manager/modules/nutanix/src/dashboard/nodes/node/general-info/routing.js [hunk 10]
```diff
@@ -0,0 +1,43 @@
+export default /* @ngInject */ ($stateProvider) => {
+  $stateProvider.state('nutanix.dashboard.nodes.node.general-info', {
+    url: '',
+    views: {
+      serverView: 'nutanixNodeGeneralInfo',
+    },
+    resolve: {
+      nodeId: /* @ngInject */ ($transition$) => $transition$.params().nodeId,
+      server: /* @ngInject */ (nodeId, Nutanix) => Nutanix.getServer(nodeId),
+      user: /* @ngInject */ (coreConfig) => coreConfig.getUser(),
```
Comment (radireddy) [inline packages/manager/modules/nutanix/src/dashboard/nodes/node/general-info/routing.js:10]: move this to nutanix root router

File: packages/manager/modules/nutanix/src/service.js [hunk 11]
```diff
@@ -0,0 +1,33 @@
...
+      },
+      data: {
+        resource: {
+          displayName,
+        },
+      },
+    });
+  }
+
+  getServer(nodeId) {
```
Comment (radireddy) [inline packages/manager/modules/nutanix/src/service.js:23]: Use a separate service for dedicated servers (nodes). Lets try to reuse the existing service.

**Review by radireddy — CHANGES_REQUESTED** [review]

File: packages/manager/modules/nutanix/src/index.js [hunk 12]
```diff
@@ -2,6 +2,8 @@ import angular from 'angular';
 import '@uirouter/angularjs';
 import 'oclazyload';
 
+import './index.scss';
```
[the remark this answers is not in the document]
Comment (varun257) [reply packages/manager/modules/nutanix/src/index.js]: I have added this to apps and have removed here

File: packages/manager/modules/nutanix/src/dashboard/nodes/node/general-info/edit-display-name/controller.js [hunk 13]
```diff
@@ -0,0 +1,62 @@
...
+
+  $onInit() {
+    this.loading = {
+      init: false,
+      update: false,
+    };
+    this.actionEnabled = true;
+
+    this.loading.init = true;
+    this.NutanixNode.getServer(this.nodeId)
```
Comment (radireddy) [inline packages/manager/modules/nutanix/src/dashboard/nodes/node/general-info/edit-display-name/controller.js:49]: why do you fetch server again? it must be fetched in node or general-info route right? can you reuse it? pass it through route resolvers.

File: packages/manager/modules/nutanix/src/dashboard/nodes/node/general-info/edit-display-name/routing.js [hunk 14]
```diff
@@ -0,0 +1,15 @@
+export default /* @ngInject */ ($stateProvider) => {
+  $stateProvider.state(
+    'nutanix.dashboard.nodes.node.general-info.edit-display-name',
+    {
+      url: '/edit-display-name',
+      component: 'nutanixNodeGeneralInfoEditDisplayName',
+      layout: 'modal',
+      resolve: {
+        goBack: /* @ngInject */ (goToNutanixNodeServer) =>
```
Comment (radireddy) [inline packages/manager/modules/nutanix/src/dashboard/nodes/node/general-info/edit-display-name/routing.js:9]:
```suggestion
        goBack: /* @ngInject */ (goToNutanixNode) =>
```

File: packages/manager/modules/nutanix/src/dashboard/nodes/node/general-info/install/service.js [hunk 15]
```diff
@@ -0,0 +1,32 @@
+import get from 'lodash/get';
+
+export default class NutanixServerInstallService {
```
Comment (radireddy) [inline packages/manager/modules/nutanix/src/dashboard/nodes/node/general-info/install/service.js:3]:
```suggestion
export default class NutanixNodeInstallService {
```

File: packages/manager/modules/nutanix/src/dashboard/nodes/node/service.js [hunk 16]
```diff
@@ -0,0 +1,33 @@
+import { DedicatedServer } from '@ovh-ux/manager-models';
+
+export default class NutanixService {
```
Comment (radireddy) [inline packages/manager/modules/nutanix/src/dashboard/nodes/node/service.js:3]:
```suggestion
export default class NutanixNodeService {
```

**Review by radireddy — CHANGES_REQUESTED** [review]

File: packages/manager/modules/nutanix/src/dashboard/nodes/node/general-info/translations/Messages_fr_FR.json (hunk 3 above)
Comment (varun257) [reply packages/manager/modules/nutanix/src/dashboard/nodes/node/general-info/translations/Messages_fr_FR.json:2]: This has to be done in `bm-server-components` and not here. As this is used in one of the components used here.
Also this means, `bm-server-components` to be rebased with latest `develop` branch as well

**Review by radireddy — APPROVED** [review]

File: packages/manager/modules/nutanix/src/dashboard/nodes/node/general-info/controller.js [hunk 17]
```diff
@@ -0,0 +1,20 @@
+export default class NutanixNodeGeneralInfoCtrl {
+  $onInit() {
+    this.loadServer();
+  }
+
+  loadServer() {
+    /* if there is no os installed, the api return "none_64" */
+    if (/^none_\d{2}?$/.test(this.server.os)) {
```
Comment (ganeshkumar1989) [inline packages/manager/modules/nutanix/src/dashboard/nodes/node/general-info/controller.js:8]: May be move the regex to a constant?

File: packages/manager/modules/nutanix/src/dashboard/nodes/node/template.html [hunk 18]
```diff
@@ -1,52 +1,56 @@
...
+            data-text="{{:: 'nutanix_node_guides_header' | translate }}"
         >
-            <span data-translate="nutanix_node_guides_header_all_guides"></span>
-        </oui-guide-menu-item>
-    </oui-guide-menu>
-</oui-header>
-<div class="my-3" data-ovh-alert="nutanix_node_alert"></div>
-<div ui-view></div>
+            <oui-guide-menu-item
+                href="{{ $ctrl.dedicatedServerGuideUrl }}"
```
Comment (ganeshkumar1989) [inline packages/manager/modules/nutanix/src/dashboard/nodes/node/template.html:45]:
```suggestion
                href="{{:: $ctrl.dedicatedServerGuideUrl }}"
```

**Review by ganeshkumar1989 — APPROVED** [review]

**Pull request merged**